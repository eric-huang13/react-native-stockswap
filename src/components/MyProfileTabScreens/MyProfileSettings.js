import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  ScrollView,
  SafeAreaView,
  Switch,
  TouchableOpacity,
  ActivityIndicator,
} from 'react-native';
import TriangleIcon from '../../icons/TriangleIcon';
import {connect} from 'react-redux';
import {Logout} from '../../actions/user';
import { GetSettings, PostSettings } from '../../actions/profile';
import {moderateScale} from '../../util/responsiveFont';
import LinearGradient from 'react-native-linear-gradient';
import {CommonActions} from '@react-navigation/native';
import PlaidComponent from './PlaidComponent';
import { Linking } from 'react-native';


class MyProfileSettings extends Component {
  constructor(props) {
    super(props);

    this.state = {
      enabled: false,
      currentEmail: '',
      shouldShow: false,
      dropDown: 'Visible for all',
    };
  }

  toggleSwitch = (value) => {
    this.setState({enabled: value});
  };
  dropDownSelect(pick) {
    this.setState({dropDown: pick, shouldShow: false});
    this.props.PostSettings({visibility: pick})
  }
openSetttings(){
  Linking.openSettings();

}
  componentDidMount() {
    const {users, userAccount, GetSettings, profileSettings} = this.props;
    GetSettings()
    this.setState({
      currentEmail: userAccount.email,
      dropDown: profileSettings.visibility
    });
  }
  render() {
    const {LogoutUser} = this.props;
    const {shouldShow} = this.state;
    if (this.props.plaidLoading) {
      return (
        <LinearGradient
          start={{x: 0.1, y: 1}}
          end={{x: 0.1, y: 0.1}}
          colors={['#1D2842', '#3d4b6e']}
          style={{flex: 1}}>
          <View style={style.loadingView}>
            <Text style={style.loadingText}>Loading...</Text>
            <ActivityIndicator size="large" color="#8b64ff" />
          </View>
        </LinearGradient>
      );
    }

    return (
      <SafeAreaView style={style.container}>
        <ScrollView>
          {/* <View style={style.topDetailsContainer}>
            <View style={style.detailsRow}>
              <View style={style.detailsColumn}>
                <Text style={style.detailsText}>Email</Text>
                <Text style={style.detailsData}>{this.state.currentEmail}</Text>
              </View>
              <TouchableOpacity
                onPress={() => this.props.navigation.navigate('ChangeEmail')}>
                <Text style={style.detailsButton}>Change</Text>
              </TouchableOpacity>
            </View>

            <View style={style.detailsRow}>
              <View style={style.detailsColumn}>
                <Text style={style.detailsText}>Phone Number</Text>
                <Text style={style.detailsData}>Test</Text>
              </View>
              <TouchableOpacity
                onPress={() =>
                  this.props.navigation.navigate('ChangePassword')
                }>
                <Text style={style.detailsButton}>Change</Text>
              </TouchableOpacity>
            </View>

            <View style={style.detailsRow}>
              <View style={style.detailsColumn}>
                <Text style={style.detailsText}>Password</Text>
                <Text style={style.detailsData}>Test</Text>
              </View>
              <View style={style.detailsButtonContainer}>
                <Text style={style.detailsButton}>Show</Text>
                <TouchableOpacity
                  style={style.detailsButtonChange}
                  onPress={() =>
                    this.props.navigation.navigate('ChangePassword')
                  }>
                  <Text style={style.detailsButton}>Change</Text>
                </TouchableOpacity>
              </View>
            </View>
          </View> */}

          <View style={style.middleContainer}>
            <View style={style.accountPrivacyContainer}>
              <Text style={style.detailsText}>Account privacy: {this.props.profileSettings.visibility}</Text>

              <View style={style.dotsDropdownContainer}>
                <TouchableOpacity
                  onPress={() =>
                    this.setState({
                      shouldShow: !shouldShow,
                    })
                  }>
                  <View style={style.visibleButtonContainer}>
                    <Text style={style.middleDetailsText}>
                      {this.state.dropDown == 'public'?
                      'Visible to all'
                      :
                      'Private'
                    }
                    </Text>
                    <TriangleIcon style={style.icon} />
                  </View>
                </TouchableOpacity>
                {this.state.shouldShow ? (
                  <View style={style.dropdown}>
                    {this.props.profileSettings.visibility == 'public' ? (
                      <TouchableOpacity
                        onPress={() => this.dropDownSelect('private')}>
                        <Text style={style.dropDownText}>Private</Text>
                      </TouchableOpacity>
                    ) : (
                      <TouchableOpacity
                        onPress={() => this.dropDownSelect('public')}>
                        <Text style={style.dropDownText}>Visible to all</Text>
                      </TouchableOpacity>
                    )}
                  </View>
                ) : null}
              </View>
            </View>
            <View style={style.notificationsContainer}>
            <TouchableOpacity
                        onPress={() => Linking.openSettings()}>
                        <Text style={style.dropDownText}>Open Settings</Text>
                      </TouchableOpacity>
              <Text style={style.middleDetailsText}>
                Turn off notifications
              </Text>
              <Switch
                trackColor={{false: '#767577', true: '#B8A0FF'}}
                thumbColor={this.state.enabled ? '#f4f3f4' : '#f4f3f4'}
                ios_backgroundColor="#f4f3f4"
                onValueChange={this.toggleSwitch}
                value={this.state.enabled}
                style={{transform: [{scaleX: 1.5}, {scaleY: 1.5}]}}
              />
            </View>
          </View>

          <View style={style.bottomContainer}>
            <View style={style.bottomInnerContiner}>
              <Text style={style.bottomText}>Privacy policy</Text>
              <TouchableOpacity
                style={style.detailsButtonChange}
                onPress={() => this.props.navigation.navigate('PrivacyPolicy')}>
                <Text style={style.detailsButton}>Show</Text>
              </TouchableOpacity>
            </View>
            <View style={style.bottomInnerContiner}>
              <Text style={style.bottomText}>Terms and conditions</Text>
              <TouchableOpacity
                style={style.detailsButtonChange}
                onPress={() =>
                  this.props.navigation.navigate('TermsAndConditions')
                }>
                <Text style={style.detailsButton}>Show</Text>
              </TouchableOpacity>
            </View>
            <View style={style.bottomInnerContiner}>
              <Text style={style.bottomText}>Accounts</Text>
              <TouchableOpacity
                style={style.detailsButtonChange}
                onPress={() =>
                  this.props.navigation.navigate('PortfolioMangement')
                }>
                <Text style={style.detailsButton}>Manage Accounts</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={style.detailsButtonChange}
                onPress={() =>
                  this.props.navigation.navigate('EnableAccounts')
                }>
                <Text style={style.detailsButton}>Enable Accounts</Text>
              </TouchableOpacity>
            </View>
            <View style={style.bottomInnerContiner}>
              <Text style={style.bottomText}>Link Bank Account</Text>

              <PlaidComponent screen={'settings'}/>
            </View>
          </View>
          <View style={style.logoutButtonContainer}>
            <Text style={style.logoutButton} onPress={() => LogoutUser()}>
              Log Out
            </Text>
          </View>
        </ScrollView>
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    user: state.user.user,
    plaidLoading: state.user.plaidLoading,
    users: state.people.users,
    userAccount: state.user.userFakeData,
    profileSettings: state.user.settings,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    LogoutUser: (userCredentials) => dispatch(Logout(userCredentials)),
    GetSettings: () => dispatch(GetSettings()),
    PostSettings: (input) => dispatch(PostSettings(input)),

  };
};

export default connect(mapStateToProps, mapDispatchToProps)(MyProfileSettings);

const style = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#2a334a',
    paddingTop: moderateScale(10),
    paddingBottom: moderateScale(20),
  },
  topDetailsContainer: {
    paddingHorizontal: moderateScale(12),
    paddingVertical: moderateScale(16),
    borderBottomWidth: moderateScale(1),
    borderBottomColor: 'rgba(158, 150, 150, .4)',
  },
  detailsRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: moderateScale(16),
  },
  detailsColumn: {},
  detailsText: {
    color: 'lightgrey',
    fontSize: moderateScale(12),
    fontFamily: 'Montserrat-Regular',
    paddingLeft: moderateScale(2),
  },
  detailsData: {
    color: '#FFFFFF',
    fontSize: moderateScale(16),
    fontFamily: 'Montserrat-SemiBold',
    marginTop: moderateScale(1),
  },
  detailsButton: {
    color: '#B8A0FF',
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-SemiBold',
    alignSelf: 'flex-end',
    marginRight: 6,
  },
  detailsButtonContainer: {
    flexDirection: 'row',
  },
  detailsButtonChange: {
    color: '#B8A0FF',
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-SemiBold',
    alignSelf: 'flex-end',
    marginLeft: 24,
  },
  middleContainer: {
    paddingHorizontal: moderateScale(12),
    paddingVertical: 30,
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(158, 150, 150, .4)',
  },
  accountPrivacyContainer: {},
  visibleButtonContainer: {
    paddingHorizontal: 10,
    paddingVertical: 10,
    borderRadius: 6,
    backgroundColor: '#3E4D6C',
    marginTop: 4,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  middleDetailsText: {
    fontFamily: 'Montserrat-Medium',
    color: '#FFFFFF',
    fontSize: moderateScale(16),
  },
  dropdown: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: 44,
    backgroundColor: '#3E4D6C',
    zIndex: 1,
    paddingVertical: 4,
    height: 35,
    position: 'absolute',
    borderBottomLeftRadius: 6,
    borderBottomRightRadius: 6,
  },
  dropDownText: {
    color: 'white',
    fontSize: moderateScale(16),
    marginHorizontal: moderateScale(12),
    fontFamily: 'Montserrat-Medium',
    marginBottom: moderateScale(6),
  },
  dropDownTextReportContainer: {
    borderTopWidth: moderateScale(1),
    borderTopColor: 'lightgrey',
    paddingTop: moderateScale(4),
    backgroundColor: '#2C3957',
  },
  icon: {
    marginRight: moderateScale(4),
  },
  notificationsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: moderateScale(36),
    marginBottom: moderateScale(10),
  },
  bottomContainer: {
    paddingVertical: moderateScale(20),
    paddingHorizontal: moderateScale(10),
    borderBottomWidth: moderateScale(1),
    borderBottomColor: 'rgba(158, 150, 150, .4)',
  },
  bottomInnerContiner: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: moderateScale(8),
    marginBottom: moderateScale(8),
  },
  bottomText: {
    fontFamily: 'Montserrat-SemiBold',
    color: '#FFFFFF',
    fontSize: moderateScale(16),
  },

  logoutButtonContainer: {
    marginTop: moderateScale(34),
    marginHorizontal: moderateScale(20),
    paddingVertical: moderateScale(10),
    borderWidth: moderateScale(1),
    borderColor: '#8B64FF',
    borderRadius: moderateScale(6),
  },
  logoutButton: {
    textAlign: 'center',
    fontFamily: 'Montserrat-SemiBold',
    color: '#8B64FF',
    fontSize: moderateScale(14),
  },
  loadingView: {
    alignContent: 'center',
    alignItems: 'center',
    marginTop: moderateScale(180),
  },
  loadingText: {
    color: '#B8A0FF',
    fontSize: moderateScale(18),
    fontFamily: 'Montserrat-SemiBold',
    marginBottom: moderateScale(24),
  },
});
