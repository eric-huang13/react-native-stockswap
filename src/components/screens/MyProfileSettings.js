import React, { Component } from 'react'
import { Text, View, StyleSheet, ScrollView, SafeAreaView,Switch, TouchableOpacity } from 'react-native'
import TriangleIcon from '../../icons/TriangleIcon'
import { connect } from "react-redux";
import {Logout} from 'actions/user';


class MyProfileSettings extends Component {
    constructor(props) {
        super(props);
    
        this.state = {
          enabled: false,
          currentEmail:'',          
        };
      }
   
    toggleSwitch = value =>{ this.setState({ enabled: value});
};    
componentDidMount() {
    const {users} = this.props
    const id = 1
    const selectedUser = users.filter((user) => user.id === id);
    {selectedUser.map((user) => {
      this.setState({
        currentEmail: user.email,        
      })
    })}
  }
    render() {
        const {LogoutUser} = this.props

        return (
            <SafeAreaView style={style.container}>
                <ScrollView>
                <View style={style.topDetailsContainer}>
                    <View style={style.detailsRow}>
                    <View style={style.detailsColumn}>
                    <Text style={style.detailsText}>Email</Text>
                    <Text style={style.detailsData}>{this.state.currentEmail}</Text>
                    </View >
                    <TouchableOpacity onPress={() => this.props.navigation.navigate('ChangeEmail')}>
                    <Text style={style.detailsButton}>Change</Text>                
                    </TouchableOpacity>
                    </View>

                    <View style={style.detailsRow}>
                    <View style={style.detailsColumn}>
                    <Text style={style.detailsText}>Phone Number</Text>
                    <Text style={style.detailsData}>Test</Text>
                    </View>
                    <TouchableOpacity onPress={() => this.props.navigation.navigate('ChangePassword')}>
                    <Text style={style.detailsButton}>Change</Text>                
                    </TouchableOpacity>
                    </View>

                    <View style={style.detailsRow}>
                    <View style={style.detailsColumn}>
                    <Text style={style.detailsText}>Password</Text>
                    <Text style={style.detailsData}>Test</Text>
                    </View>
                    <View style={style.detailsButtonContainer}>
                    <Text style={style.detailsButton}>Show</Text>
                    <TouchableOpacity style={style.detailsButtonChange} onPress={() => this.props.navigation.navigate('ChangePassword')}>
                    <Text style={style.detailsButton}>Change</Text>
                    </TouchableOpacity>

                    </View>
                    </View>

                </View>

                <View style={style.middleContainer}>
                <View style={style.accountPrivacyContainer}>
                    <Text style={style.detailsText}>Account privacy</Text>
                    <View style={style.visibleButtonContainer}>
                        <Text style={style.middleDetailsText}>Visible for all</Text>
                        <TriangleIcon/>
                    </View>
                </View>
                <View style={style.notificationsContainer}>
                    <Text style={style.middleDetailsText}>Turn off notifications</Text>
                    <Switch 
        trackColor={{ false: "#767577", true: "#B8A0FF" }}
        thumbColor={this.state.enabled ? "#f4f3f4" : "#f4f3f4"}
        ios_backgroundColor="#f4f3f4"
        onValueChange={this.toggleSwitch}
        value={this.state.enabled}
        style={{ transform: [{ scaleX: 1.5 }, { scaleY: 1.5 }] }}
       
      />
                </View>

                </View>
                
                <View style={style.bottomContainer}>
                <View style={style.bottomInnerContiner}>
                    <Text style={style.bottomText}>Privacy policy</Text>
                    <Text style={style.detailsButton}>Show</Text>
                </View>
                <View style={style.bottomInnerContiner}>
                    <Text style={style.bottomText}>Terms and conditions</Text>
                    <Text style={style.detailsButton}>Show</Text>
                </View>

                </View>
                <View style={style.logoutButtonContainer}>
                <Text style={style.logoutButton} onPress={() => LogoutUser()}>Log Out</Text>
                </View>
                </ScrollView>
            </SafeAreaView>
        )
    }
}

const mapStateToProps = (state) => {
    return {
      user: state.user.user,
      users: state.company.users,
    
    };
  };
  
  const mapDispatchToProps = (dispatch) => {
    return {
        LogoutUser: (userCredentials) => dispatch(Logout(userCredentials)),

    };
  };
  
  export default connect(mapStateToProps, mapDispatchToProps)(MyProfileSettings);

const style = StyleSheet.create({
    container: {
    flex: 1,
    backgroundColor: '#2a334a',
    paddingTop:10,
    paddingBottom:20,
    },
    topDetailsContainer:{
    paddingHorizontal:12,
    paddingVertical:16,
    borderBottomWidth:1,
    borderBottomColor: 'rgba(158, 150, 150, .4)',

    },
    detailsRow:{
    flexDirection:'row',
    justifyContent:'space-between',
    paddingVertical:16,
    },
    detailsColumn:{

    },
    detailsText:{
    color: 'lightgrey',
    fontSize: 12,
    fontFamily:'Montserrat-Regular', 
    paddingLeft:2,
          
    },
    detailsData:{
    color: '#FFFFFF',
    fontSize: 16,
    fontFamily:'Montserrat-SemiBold',
    marginTop:1,

    },
    detailsButton:{
    color: '#B8A0FF',
    fontSize: 14,
    fontFamily:'Montserrat-SemiBold',
    alignSelf:'flex-end',
    marginRight:6,

    },
    detailsButtonContainer:{
    flexDirection:'row',

    },
    detailsButtonChange:{
    color: '#B8A0FF',
    fontSize: 14,
    fontFamily:'Montserrat-SemiBold',
    alignSelf:'flex-end',
    marginLeft:24,  
    },
    middleContainer:{
    paddingHorizontal:12,
    paddingVertical:30,
    borderBottomWidth:1,
    borderBottomColor: 'rgba(158, 150, 150, .4)',

    },
    accountPrivacyContainer:{
    },
    visibleButtonContainer:{
        paddingHorizontal:10,
        paddingVertical:10,
        borderRadius:6,
        backgroundColor:'#3E4D6C',
        marginTop:4,
        flexDirection:'row',
        justifyContent:'space-between',
        alignItems:'center'
        
    },
    middleDetailsText:{
    fontFamily:'Montserrat-Medium',
    color:'#FFFFFF',
    fontSize:16,
    
        
    },
    notificationsContainer:{
        flexDirection:'row',
        justifyContent:'space-between',
        marginTop:36,
        marginBottom:10,

    },
    bottomContainer:{
        paddingVertical:20,
        paddingHorizontal:10,
        borderBottomWidth:1,
        borderBottomColor: 'rgba(158, 150, 150, .4)',
  
    },
    bottomInnerContiner:{
        flexDirection:'row',
        justifyContent:'space-between',
        paddingVertical:8,
        marginBottom:8,
    },
    bottomText:{
    fontFamily:'Montserrat-SemiBold',
    color:'#FFFFFF',
    fontSize:16,

    },
    
    logoutButtonContainer:{
    marginTop:34,
    marginHorizontal:20,
    paddingVertical:10,
    borderWidth:1,
    borderColor:'#8B64FF',
    borderRadius:6,    
   },
    logoutButton:{
    textAlign:'center',
    fontFamily:'Montserrat-SemiBold',
    color:'#8B64FF',
    fontSize:14,


    },
})      