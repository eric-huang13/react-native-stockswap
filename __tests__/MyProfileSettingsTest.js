import React from 'react';
import MyProfileSettings from '../src/components/MyProfileTabScreens/MyProfileSettings';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6

jest.mock('@react-native-community/google-signin', () => {});

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<MyProfileSettings />);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});