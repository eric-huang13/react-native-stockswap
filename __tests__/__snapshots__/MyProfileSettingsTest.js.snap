// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
      "paddingBottom": 28.248175182481752,
      "paddingTop": 14.124087591240876,
    }
  }
>
  <ScrollView>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(158, 150, 150, .4)",
          "borderBottomWidth": 1.4124087591240877,
          "paddingHorizontal": 16.94890510948905,
          "paddingVertical": 22.598540145985403,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 22.598540145985403,
          }
        }
      >
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "color": "lightgrey",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16.94890510948905,
                "paddingLeft": 2.8248175182481754,
              }
            }
          >
            Email
          </Text>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 22.598540145985403,
                "marginTop": 1.4124087591240877,
              }
            }
          >
            
          </Text>
        </View>
        <ForwardRef
          onPress={[Function]}
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#B8A0FF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 19.77372262773723,
                "marginRight": 6,
              }
            }
          >
            Change
          </Text>
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 22.598540145985403,
          }
        }
      >
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "color": "lightgrey",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16.94890510948905,
                "paddingLeft": 2.8248175182481754,
              }
            }
          >
            Phone Number
          </Text>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 22.598540145985403,
                "marginTop": 1.4124087591240877,
              }
            }
          >
            Test
          </Text>
        </View>
        <ForwardRef
          onPress={[Function]}
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#B8A0FF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 19.77372262773723,
                "marginRight": 6,
              }
            }
          >
            Change
          </Text>
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingVertical": 22.598540145985403,
          }
        }
      >
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "color": "lightgrey",
                "fontFamily": "Montserrat-Regular",
                "fontSize": 16.94890510948905,
                "paddingLeft": 2.8248175182481754,
              }
            }
          >
            Password
          </Text>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 22.598540145985403,
                "marginTop": 1.4124087591240877,
              }
            }
          >
            Test
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#B8A0FF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 19.77372262773723,
                "marginRight": 6,
              }
            }
          >
            Show
          </Text>
          <ForwardRef
            onPress={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#B8A0FF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 19.77372262773723,
                "marginLeft": 24,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "color": "#B8A0FF",
                  "fontFamily": "Montserrat-SemiBold",
                  "fontSize": 19.77372262773723,
                  "marginRight": 6,
                }
              }
            >
              Change
            </Text>
          </ForwardRef>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(158, 150, 150, .4)",
          "borderBottomWidth": 1,
          "paddingHorizontal": 16.94890510948905,
          "paddingVertical": 30,
        }
      }
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Object {
              "color": "lightgrey",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 16.94890510948905,
              "paddingLeft": 2.8248175182481754,
            }
          }
        >
          Account privacy
        </Text>
        <View>
          <ForwardRef
            onPress={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#3E4D6C",
                  "borderRadius": 6,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginTop": 4,
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Montserrat-Medium",
                    "fontSize": 22.598540145985403,
                  }
                }
              >
                Visible for all
              </Text>
              <TriangleIcon
                style={
                  Object {
                    "marginRight": 5.649635036496351,
                  }
                }
              />
            </View>
          </ForwardRef>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 14.124087591240876,
            "marginTop": 50.846715328467155,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Medium",
              "fontSize": 22.598540145985403,
            }
          }
        >
          Turn off notifications
        </Text>
        <Switch
          ios_backgroundColor="#f4f3f4"
          onValueChange={[Function]}
          style={
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1.5,
                },
                Object {
                  "scaleY": 1.5,
                },
              ],
            }
          }
          thumbColor="#f4f3f4"
          trackColor={
            Object {
              "false": "#767577",
              "true": "#B8A0FF",
            }
          }
          value={false}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": "rgba(158, 150, 150, .4)",
          "borderBottomWidth": 1.4124087591240877,
          "paddingHorizontal": 14.124087591240876,
          "paddingVertical": 28.248175182481752,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 11.299270072992702,
            "paddingVertical": 11.299270072992702,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-SemiBold",
              "fontSize": 22.598540145985403,
            }
          }
        >
          Privacy policy
        </Text>
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#B8A0FF",
              "fontFamily": "Montserrat-SemiBold",
              "fontSize": 19.77372262773723,
              "marginLeft": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#B8A0FF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 19.77372262773723,
                "marginRight": 6,
              }
            }
          >
            Show
          </Text>
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 11.299270072992702,
            "paddingVertical": 11.299270072992702,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-SemiBold",
              "fontSize": 22.598540145985403,
            }
          }
        >
          Terms and conditions
        </Text>
        <ForwardRef
          onPress={[Function]}
          style={
            Object {
              "alignSelf": "flex-end",
              "color": "#B8A0FF",
              "fontFamily": "Montserrat-SemiBold",
              "fontSize": 19.77372262773723,
              "marginLeft": 24,
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#B8A0FF",
                "fontFamily": "Montserrat-SemiBold",
                "fontSize": 19.77372262773723,
                "marginRight": 6,
              }
            }
          >
            Show
          </Text>
        </ForwardRef>
      </View>
    </View>
    <View
      style={
        Object {
          "borderColor": "#8B64FF",
          "borderRadius": 8.474452554744525,
          "borderWidth": 1.4124087591240877,
          "marginHorizontal": 28.248175182481752,
          "marginTop": 48.021897810218974,
          "paddingVertical": 14.124087591240876,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "color": "#8B64FF",
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 19.77372262773723,
            "textAlign": "center",
          }
        }
      >
        Log Out
      </Text>
    </View>
  </ScrollView>
</ForwardRef(SafeAreaView)>
`;
